lib/Controller/PathController.php

```diff
@@ -4,89 +4,319 @@
+	 */
+	public static function get($dir, $files, $params = null) {
+		$view = \OC\Files\Filesystem::getView();
+		$getType = self::FILE;
+		try {
+			$filename = $dir . '/' . $files;
+			if (!$view->is_dir($filename)) {
+				self::getSingleFile($view, $dir, $files, is_null($params) ? array() : $params);
+				return;
+			}
+			set_time_limit($executionTime);
+			self::unlockAllTheFiles($dir, $files, $getType, $view, $filename);
+		} catch (\OCP\Lock\LockedException $ex) {
+			self::unlockAllTheFiles($dir, $files, $getType, $view, $filename);
+			OC::$server->getLogger()->logException($ex);
+			$l = \OC::$server->getL10N('core');
+			$hint = method_exists($ex, 'getHint') ? $ex->getHint() : '';
+			\OC_Template::printErrorPage($l->t('File is currently busy, please try again later'), $hint);
+		} catch (\OCP\Files\ForbiddenException $ex) {
+			self::unlockAllTheFiles($dir, $files, $getType, $view, $filename);
+			OC::$server->getLogger()->logException($ex);
+			$l = \OC::$server->getL10N('core');
+			\OC_Template::printErrorPage($l->t('Can\'t read file'), $ex->getMessage());
+		} catch (\Exception $ex) {
+			self::unlockAllTheFiles($dir, $files, $getType, $view, $filename);
+			OC::$server->getLogger()->logException($ex);
+			$l = \OC::$server->getL10N('core');
+			$hint = method_exists($ex, 'getHint') ? $ex->getHint() : '';
+			\OC_Template::printErrorPage($l->t('Can\'t read file'), $hint);
+		}
+	}
+
+	/**
+	 * CAUTION: the @Stuff turns off security checks; for this page no admin is
+	 *          required and no CSRF check. If you don't know what CSRF is, read
+	 *          it up in the docs or you might create a security hole. This is
+	 *          basically the only required method to add this exemption, don't
+	 *          add it to any other method if you don't exactly know what it does
+	 *
+	 * @PublicPage
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function index($uid, $path)
+	{
+		// check user exist
+		$user = $this->userManager->get($uid);
+		if (! $user) {
+			http_response_code(404);
+			exit;
+		}
+
+		$userFolder = $this->rootFolder->getUserFolder($uid);
+		try {
+			$file = $userFolder->get($path);
+
+			// check share permission
+			$shares = $this->shareManager->getSharesBy($uid, Share::SHARE_TYPE_LINK, $file);
+			$share  = $shares[0] ?? null;
+			if (! $share ||
+				$share->getPassword() ||
+				($share->getExpirationDate() && $share->getExpirationDate()->getTimestamp() < time())) {
+				http_response_code(403);
+				exit;
+			}
+
+			// todo version file handle
+
+			$path = $userFolder->getRelativePath($file->getPath());
+
+			// output file contents without Content-Disposition header
+			//            header('Content-Transfer-Encoding: binary', true);
+			//            header('Pragma: public');
+			//            header('Expires: 0');
+			//            header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+
+			\OC_Util::setupFS($uid);
+			$dir = dirname($path);
+			$files = basename($path);
```

Comment:
no need pass $files

Reply:
Hi, still new to the github thingy ;) I just commited fixes to the pull request, please review. This point should be fixed (among other things).